src/MongoDB.Driver.Core/Core/Compression/Zstd/NativeBufferInfo.cs

```diff
@@ -0,0 +1,57 @@
+﻿/* Original work:
```

Comment:
I took this as as example: https://github.com/DmitryLukyanov/mongo-csharp-driver/blob/eaaf72f7c50095272d9473ef4d4e6de0a6e24cec/src/MongoDB.Driver.Core/Core/Authentication/Vendored/Rfc2898DeriveBytes.cs#L1-L42